dali/operators/reader/nvdecoder/dynlink_nvcuvid.cc

```diff
@@ -56,52 +57,55 @@ tcuvidCtxUnlock                       ptr_cuvidCtxUnlock;
 
 #include <dlfcn.h>
 
+typedef void *DLLDRIVER;
+
 static char __DriverLibName[] = "libnvcuvid.so";
 static char __DriverLibName1[] = "libnvcuvid.so.1";
-
-typedef void *DLLDRIVER;
+static DLLDRIVER DriverLib = NULL;
+static std::mutex m;
+static std::atomic_bool initialized(false);
```

Comment:
AFAIK this does not to bee atomic, because we only modify it when owning the mutex `m`.

Reply:
Done